src/NuGet.Core/NuGet.DependencyResolver.Core/Remote/RemoteDependencyWalker.cs

```diff
@@ -75,41 +75,10 @@ public async Task<GraphNode<RemoteResolveResult>> WalkAsync(LibraryRange library
             bool hasParentNodes)
         {
             HashSet<LibraryDependency> runtimeDependencies = null;
-            List<Task<GraphNode<RemoteResolveResult>>> tasks = null;
 
             if (runtimeGraph != null && !string.IsNullOrEmpty(runtimeName))
             {
-                // HACK(davidfowl): This is making runtime.json support package redirects
-
-                // Look up any additional dependencies for this package
-                foreach (var runtimeDependency in runtimeGraph.FindRuntimeDependencies(runtimeName, libraryRange.Name))
-                {
-                    var libraryDependency = new LibraryDependency
-                    {
-                        LibraryRange = new LibraryRange()
-                        {
-                            Name = runtimeDependency.Id,
-                            VersionRange = runtimeDependency.VersionRange,
-                            TypeConstraint = LibraryDependencyTarget.PackageProjectExternal
-                        }
-                    };
-
-                    if (StringComparer.OrdinalIgnoreCase.Equals(runtimeDependency.Id, libraryRange.Name))
-                    {
-                        if (libraryRange.VersionRange != null &&
-                            runtimeDependency.VersionRange != null &&
-                            libraryRange.VersionRange.MinVersion < runtimeDependency.VersionRange.MinVersion)
-                        {
-                            libraryRange = libraryDependency.LibraryRange;
-                        }
-                    }
-                    else
-                    {
-                        // Otherwise it's a dependency of this node
-                        runtimeDependencies ??= new HashSet<LibraryDependency>(LibraryDependencyNameComparer.OrdinalIgnoreCaseNameComparer);
-                        runtimeDependencies.Add(libraryDependency);
-                    }
-                }
+                EvaluateRuntimeDependencies(ref libraryRange, runtimeName, runtimeGraph, ref runtimeDependencies);
```

Comment:
This is just a refactoring correct?

Reply:
Correct. I can revert this. It was helpful while investigating.

Reply:
I don't care too much.
Sometimes we like keeping history because it makes blame more valuable, but we've done a lot of refactoring in this area. 

As far as I'm concerned, it's ok as is.